PROTO_tests/tests/numeric_test.py

```diff
@@ -0,0 +1,362 @@
+        elif num_type == ak.float64:
+            strarr = ak.array(["1.1", "2.2 ", "3?.3", "4.!4", "  5.5", "6.6e-6", "78.91E+4", "6", "N/A"])
+            ans = np.array([1.1, 2.2, np.nan, np.nan, 5.5, 6.6e-6, 78.91e4, 6.0, np.nan])
+        elif num_type == ak.bool:
+            strarr = ak.array(
+                ["True", "False ", "Neither", "N/A", "  True", "true", "false", "TRUE", "NOTTRUE"]
+            )
+            ans = np.array([True, False, False, False, True, True, False, True, False])
+
+        validans = ak.array([True, True, False, False, True, True, True, True, False])
+
+        with pytest.raises(RuntimeError):
+            ak.cast(strarr, num_type, errors=ak.ErrorMode.strict)
+        res = ak.cast(strarr, num_type, errors=ak.ErrorMode.ignore)
+        assert np.allclose(ans, res.to_ndarray(), equal_nan=True)
+        res, valid = ak.cast(strarr, num_type, errors=ak.ErrorMode.return_validity)
+        assert valid.to_list() == validans.to_list()
+        assert np.allclose(ans, res.to_ndarray(), equal_nan=True)
+
+    def test_histogram(self):
+        pda = ak.randint(10, 30, 40)
+        bins, result = ak.histogram(pda, bins=20)
+
+        assert isinstance(result, ak.pdarray)
+        assert 20 == len(bins)
+        assert 20 == len(result)
+        assert int == result.dtype
+
+        with pytest.raises(TypeError):
+            ak.histogram([range(0, 10)], bins=1)
+
+        with pytest.raises(TypeError):
+            ak.histogram(pda, bins="1")
+
+        with pytest.raises(TypeError):
+            ak.histogram([range(0, 10)], bins="1")
+
+    def test_log(self):
+        na = np.linspace(1, 10, 10)
+        pda = ak.array(na)
+
+        assert np.allclose(np.log(na), ak.log(pda).to_ndarray())
+        with pytest.raises(TypeError):
+            ak.log([range(0, 10)])
+
+    def test_exp(self):
+        na = np.linspace(1, 10, 10)
+        pda = ak.array(na)
+
+        assert np.allclose(np.exp(na), ak.exp(pda).to_ndarray())
+        with pytest.raises(TypeError):
+            ak.exp([range(0, 10)])
+
+    def test_abs(self):
+        na = np.linspace(1, 10, 10)
+        pda = ak.array(na)
+
+        assert np.allclose(np.abs(na), ak.abs(pda).to_ndarray())
+        assert ak.arange(5, 1, -1).to_list() == ak.abs(ak.arange(-5, -1)).to_list()
+        assert (
+            [5, 4, 3, 2, 1]
+            == ak.abs(ak.linspace(-5, -1, 5)).to_list(),
+        )
+
+        with pytest.raises(TypeError):
+            ak.abs([range(0, 10)])
+
+    def test_cumsum(self):
+        na = np.linspace(1, 10, 10)
+        pda = ak.array(na)
+
+        assert np.allclose(np.cumsum(na), ak.cumsum(pda).to_ndarray())
+
+        # Test uint case
+        na = np.linspace(1, 10, 10, "uint64")
+        pda = ak.cast(pda, ak.uint64)
+
+        assert np.allclose(np.cumsum(na), ak.cumsum(pda).to_ndarray())
+
+        with pytest.raises(TypeError):
+            ak.cumsum([range(0, 10)])
```

Comment:
definitely this one (and likely some of the others) could be parameterized on `NUMERIC_TYPES`. I'm not sure if we wanna do that in this pass or not

maybe something like:
```python
@pytest.mark.parametrize("num_type", NUMERIC_TYPES)
def test_cumsum(self):
    na = np.arange(1, 10, dtype=num_type)
    pda = ak.array(na)

    assert np.allclose(np.cumsum(na), ak.cumsum(pda).to_ndarray())
```

Reply:
I parameterized the other functions I thought made sense. I added a few other lists to use based on the typing that was valid for the functions being tested.